### PR TITLE
Docs: add QAOA best practices guide

### LINKED ISSUE #33: Docs: Add a QAOA best-practices guide for QiskitOpt.jl users
Parent: #32

## Summary

Add a QAOA best-practices guide that translates the relevant IBM community guidance and arXiv:2606.05311 recommendations into practical QiskitOpt.jl usage.

## Content to cover

- Start local with Aer and deterministic seeds before moving to IBM hardware.
- Use existing QAOA parameter metadata to record the source, names, and values of initial parameters.
- Explain available initializer choices: zero default, random seeded starts, fixed Wurtz-Lykov angles, and the schedule helpers tracked in #34.
- Explain when hardware-aware transpilation belongs in upstream qopt-best-practices and how QiskitOpt.jl should expose only an adapter hook.
- Explain that Pauli propagation, MPS angle training, SAT mapping, SWAP strategies, qubit selection, and CVaR training loops are generic QAOA capabilities and should not be reimplemented here.

## Acceptance criteria

- A new documentation page or README section exists and is linked from the main README.
- The guide includes links to qopt-best-practices, qaoa_training_pipeline, and arXiv:2606.05311.
- The guide names the existing QiskitOpt.jl attributes a user should set for local Aer, IBM hardware, QAOA depth, reads, initial parameters, and seeds.
- The guide makes the upstream boundary explicit.

### PR DESCRIPTION
**Pull request by bernalde**
Summary
- Add `docs/qaoa_best_practices.md` with a practical QAOA workflow for QiskitOpt.jl users.
- Link the guide from the main README.
- Document the upstream boundary for hardware-aware transpilation, training pipelines, MPS/Pauli propagation, SAT mapping, SWAP strategies, qubit selection, and CVaR loops.

Tests
- `git diff --check`
- `julia --project=. -e 'using Pkg; Pkg.test()'`

Closes #33

### CODE REVIEW

**Review by bernalde — COMMENTED** [review]
No actionable issues found.

Blocking issues: none.
Nonblocking issues: none.
Questions: none.

Tests run and outcomes:
- `git diff --check origin/main...pr-39-review`: passed.
- `julia --project=. -e 'using Pkg; Pkg.test()'`: passed locally.
- GitHub CI: Julia 1.10 on Ubuntu, Julia 1 on Ubuntu, and Julia 1 on Windows all passed.

Merge-readiness:
The merge-base diff matches GitHub's reported scope: 2 files changed, 178 additions, 0 deletions. The PR satisfies issue #33's requested README link, guide links, attribute coverage, initializer coverage, and upstream-boundary documentation. The PR is still draft; aside from draft status and requiring formal approval from a non-author maintainer, I found no blockers.

Because I authored this PR, I am submitting this as COMMENT rather than APPROVE. The formal approval must come from another maintainer.

**Review by bernalde — COMMENTED** [review]
No actionable issues found.

Blocking: none.
Nonblocking: none.
Question: none.

Tests run and outcomes:
- `gh pr diff 39 --patch`: inspected.
- `gh api repos/JuliaQUBO/QiskitOpt.jl/pulls/39/files --paginate`: reported 2 changed files, 178 additions, 0 deletions.
- `git diff --numstat origin/main...refs/remotes/origin/pr-39-head`: matched GitHub's reported file and line totals.
- `git diff --check origin/main...HEAD`: passed.
- `gh issue view 33 --json number,title,body,state,author,labels,assignees,comments,url`: confirmed the PR satisfies the motivating issue's guide, link, attribute, initializer, and upstream-boundary requests.
- `gh issue view 34 --json number,title,body,state,author,labels,comments,url`: confirmed the referenced schedule-helper follow-up exists and matches the guide's scope.
- `gh api repos/qiskit-community/qopt-best-practices`: resolved.
- `gh api repos/qiskit-community/qaoa_training_pipeline`: resolved.
- `julia --project=. -e 'using Pkg; Pkg.test()'`: passed locally from a detached clean checkout at the PR head.
- `gh pr checks 39`: CI passed on Julia 1.10 Ubuntu, Julia 1 Ubuntu, and Julia 1 Windows.

Merge-readiness:
The PR is not draft, CI is green, and the docs change matches issue #33. Since I am the PR author, I am submitting this as COMMENT rather than APPROVE. Formal approval must come from another maintainer.

File: docs/qaoa_best_practices.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+| Final sampling shots | `QiskitOpt.QUBODrivers.FinalNumberOfReads()` |
+| Optimizer iterations | `QiskitOpt.QAOA.MaximumIterations()` |
+| Initial angle values | `QiskitOpt.QAOA.InitialParameters()` |
+| Initial angle provenance | `QiskitOpt.QAOA.InitialParameterSource()` |
+| Initial metadata recording | `QiskitOpt.QAOA.RecordInitialParameters()` |
+| Local Aer method | `QiskitOpt.QAOA.AerBackendMethod()` |
+| Local Aer precision | `QiskitOpt.QAOA.AerPrecision()` |
+| Local Aer simulator seed | `QiskitOpt.QAOA.AerSeedSimulator()` |
+| Qiskit transpiler seed | `QiskitOpt.QAOA.TranspilerSeed()` |
+| Aer MPS controls | `AerMPSOmpThreads()`, `AerMPSTruncationThreshold()`, `AerMPSMaxBondDimension()`, `AerMPSSampleMeasureAlgorithm()` |
```
Comment (bernalde) [inline docs/qaoa_best_practices.md]: Nonblocking: every other row in this table uses the fully qualified `QiskitOpt.QAOA.` prefix, but this row lists bare `AerMPSOmpThreads()`, `AerMPSTruncationThreshold()`, `AerMPSMaxBondDimension()`, `AerMPSSampleMeasureAlgorithm()`. A user copy-pasting these as written would get an UndefVarError. Prefix them with `QiskitOpt.QAOA.` to match the rest of the table. (I confirmed all four exist under `QAOA`.)

File: docs/qaoa_best_practices.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
+# QAOA Best Practices
+
+This guide maps the QAOA workflow guidance from
+[qopt-best-practices](https://github.com/qiskit-community/qopt-best-practices),
+[qaoa_training_pipeline](https://github.com/qiskit-community/qaoa_training_pipeline),
+and [arXiv:2606.05311](https://arxiv.org/abs/2606.05311) onto the
```
Comment (bernalde) [inline docs/qaoa_best_practices.md:6]: Question: this cites `arXiv:2606.05311` (carried over from issue #33). I could not verify from here that the ID resolves to the intended paper. Please confirm it is correct and the link works before merge — a dangling citation in a best-practices guide undermines its authority. Note the in-code `FIXED_ANGLE_SOURCE` cites a different paper (arXiv:2107.00677, Wurtz-Lykov), which is expected; this is only about the 2606.05311 reference.

**Review by bernalde — COMMENTED** [review]
Review as maintainer. Note: I am the PR author, so GitHub only permits a COMMENT event here; the formal approval or change request must come from another maintainer. (Two prior COMMENTED reviews of mine exist on this PR; this is a fresh, fuller pass.) No blocking issues.

Setup: fetched base (origin/main, merge-base 4b30402) and head e20292d with git; diff against the merge-base matches the PR's reported totals (2 files, +178/-0). Reviewed the head in a clean worktree. This is a docs-only change (a new `docs/qaoa_best_practices.md` plus a README link); no code paths are touched.

Intent vs issue #33: acceptance criteria are met. The guide exists and is linked from the README; it links qopt-best-practices, qaoa_training_pipeline, and arXiv:2606.05311; it names the QiskitOpt.jl attributes for local Aer, IBM hardware, QAOA depth, reads, initial parameters, and seeds; it covers all four initializer choices (zero default, seeded random, fixed Wurtz-Lykov, schedule helpers tracked in #34); and it makes the upstream boundary explicit. `Closes #33` is appropriate.

Because this is documentation, I verified the API claims by execution rather than by reading:
- Every attribute/helper the guide names exists and is callable: NumberOfLayers, NumberOfReads, QUBODrivers.FinalNumberOfReads, MaximumIterations, InitialParameters, InitialParameterSource, RecordInitialParameters, AerBackendMethod, AerPrecision, AerSeedSimulator, TranspilerSeed, the four AerMPS* controls, IBMBackend, IsLocal, Channel, Instance, IBMFakeBackend.
- `random_initial_parameters(number_of_layers=2, seed=73001)` and `fixed_angle_initial_parameters(number_of_layers=2, gamma_sign=-1)` run with the documented keyword signatures; `FIXED_ANGLE_SOURCE` resolves.
- The documented parameter order (all beta then all gamma) is correct: `parameter_names(...; number_of_layers=2)` returns ["β[0]","β[1]","γ[0]","γ[1]"].
- The default Aer method is `matrix_product_state`, matching the guide's claim that a first run needs no IBM token.
The JuMP-based example matches the existing README convention (the main README already uses `using JuMP; Model(QiskitOpt.QAOA.Optimizer)`), so JuMP as a user-side dependency is consistent with established docs.

Blocking issues:
- None.

Nonblocking issues:
1. Inconsistent qualifier in the attributes table. Every row uses the fully qualified `QiskitOpt.QAOA.` form except the Aer MPS controls row, which lists bare `AerMPSOmpThreads()`, `AerMPSTruncationThreshold()`, `AerMPSMaxBondDimension()`, `AerMPSSampleMeasureAlgorithm()`. A reader copy-pasting those bare names hits an UndefVarError. Prefix them with `QiskitOpt.QAOA.` for consistency. See inline comment.

Questions:
1. arXiv citation. The guide cites `arXiv:2606.05311` (transcribed from issue #33). I cannot verify from here that this ID resolves to the intended paper, and a wrong/dangling citation in a best-practices guide is a credibility issue. Can you confirm the ID is correct and the link resolves? See inline comment.

Tests run (clean worktree at e20292d):
- `julia --project=. -e 'using Pkg; Pkg.instantiate(); Pkg.test()'` — passed (docs-only change, suite unaffected; ran it to confirm).
- API verification script described above — all documented symbols exist and behave as written.

Merge-readiness: no blocking issues. The single nonblocking qualifier nit and the citation question are worth addressing but do not block merge on their own.

**Comment by bernalde** [comment]
Addressed review comments in commit c0635ea:
- `c0635ea` docs: qualify QAOA Aer MPS attributes

Main changes:
- Updated the Aer MPS controls row in `docs/qaoa_best_practices.md` to use the fully qualified `QiskitOpt.QAOA.*` attribute names.
- Verified `https://arxiv.org/abs/2606.05311` resolves with HTTP 200 and the page title is `Setting angles in quantum approximate optimization at utility-scale`; no documentation change was needed for that thread.

Tests run and results:
- `curl -I -L https://arxiv.org/abs/2606.05311`: passed, HTTP 200.
- `curl -L https://arxiv.org/abs/2606.05311 -o /tmp/arxiv-2606.05311.html`: passed; title and citation metadata match the intended paper.
- `git diff --check`: passed.
- `julia --project=. -e 'using QiskitOpt; attrs = (QiskitOpt.QAOA.AerMPSOmpThreads, QiskitOpt.QAOA.AerMPSTruncationThreshold, QiskitOpt.QAOA.AerMPSMaxBondDimension, QiskitOpt.QAOA.AerMPSSampleMeasureAlgorithm); println(length(attrs))'`: passed.
- `julia --project=. -e 'using Pkg; Pkg.test()'`: passed.
- `gh pr checks 39`: all CI jobs passed for the pushed commit.

Comments intentionally not addressed:
- None. The arXiv thread was addressed by verification rather than a file change.

Remaining risks or follow-up items:
- None identified.

File: docs/qaoa_best_practices.md (hunk 1 above)
Comment (bernalde) [reply docs/qaoa_best_practices.md]: Addressed in https://github.com/JuliaQUBO/QiskitOpt.jl/commit/c0635ea. The Aer MPS controls row now uses the fully qualified `QiskitOpt.QAOA.*` names, and I verified the four attributes resolve in Julia.

File: docs/qaoa_best_practices.md (hunk 2 above)
Comment (bernalde) [reply docs/qaoa_best_practices.md:6]: Addressed by verification. `curl -I -L https://arxiv.org/abs/2606.05311` returns HTTP 200, and the fetched page title/citation metadata identify `Setting angles in quantum approximate optimization at utility-scale`, so the existing citation is correct. No file change was needed.
